neqo-transport/src/fc.rs

```diff
@@ -239,6 +239,11 @@ where
         self.max_data = new_max;
         self.frame_pending = false;
     }
+
+    pub fn change_max_stream_data(&mut self, max_data: u64) {
+        self.frame_pending |= self.max_active < max_data;
```

Comment:
It seems to me like this is going to send MAX_STREAM_DATA sooner than we would otherwise.  I guess that's good, but you should document that intent.